pkg/odo/cli/service/create.go

```diff
@@ -45,46 +49,96 @@ func NewServiceCreateOptions() *ServiceCreateOptions {
 
 // Complete completes ServiceCreateOptions after they've been created
 func (o *ServiceCreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	o.serviceType = args[0]
-	// if only one arg is given, then it is considered as service name and service type both
-	o.serviceName = o.serviceType
-	// if two args are given, first is service type and second one is service name
-	if len(args) == 2 {
-		o.serviceName = args[1]
+	if len(args) == 0 || !cmd.HasFlags() {
+		o.interactive = true
 	}
+
 	o.Context = genericclioptions.NewContextCreatingAppIfNeeded(cmd)
-	return err
+
+	client := o.Client
+
+	var class scv1beta1.ClusterServiceClass
+	if o.interactive {
+		classesByCategory, _ := client.GetServiceClassesByCategory()
+		class, o.serviceType = ui.SelectClassInteractively(classesByCategory)
+
+		plans, _ := client.GetMatchingPlans(class)
+
+		var svcPlan scv1beta1.ClusterServicePlan
+		// if there is only one available plan, we select it
+		if len(plans) == 1 {
+			for k, v := range plans {
+				o.plan = k
+				svcPlan = v
+			}
+			glog.V(4).Infof("Plan %s was automatically selected since it's the only one available for service %s", o.plan, o.serviceType)
+		} else {
+			// otherwise select the plan interactively
+			o.plan = ui.SelectPlanNameInteractively(plans, "Which service plan should we use ")
+			svcPlan = plans[o.plan]
+		}
+
+		o.parametersMap = ui.EnterServicePropertiesInteractively(svcPlan)
+		o.serviceName = ui.EnterServiceNameInteractively(o.serviceType, "How should we name your service ", validateNameFn)
+	} else {
+		o.serviceType = args[0]
+		// if only one arg is given, then it is considered as service name and service type both
+		o.serviceName = o.serviceType
+		// if two args are given, first is service type and second one is service name
+		if len(args) == 2 {
+			o.serviceName = args[1]
+		}
+	}
+
+	return
+}
+
+var validateNameFn = func(i interface{}) error {
```

Comment:
I think this code should be in some utils file. Also why not use the two lines directly ( function seems to be used only once)?

Reply:
See #1126

Reply:
ok :+1: